Added testcases imageclassification model

KonakanchiSwathi opened the pull request:
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/KonakanchiSwathi/mlflow/pull/10167?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/10167/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 10167
```

</p>
</details>

### Related Issues/PRs

<!-- Uncomment 'Resolve' if this PR can close the linked items. -->
<!-- Resolve --> #xxx

### What changes are proposed in this pull request?

<!-- Please fill in changes proposed in this PR. -->

### How is this PR tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests

<!-- Attach code, screenshot, video used for manual testing here. -->

### Does this PR require documentation update?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

### Release Notes

#### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

<!-- Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change. -->

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/gateway`: AI Gateway service, Gateway client APIs, third-party Gateway integrations
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [x] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [x] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<!--
Insert an empty named anchor here to allow jumping to this section with a fragment URL
(e.g. https://github.com/mlflow/mlflow/pull/123#user-content-release-note-category).
Note that GitHub prefixes anchor names in markdown with "user-content-".
-->

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

KonakanchiSwathi committed:
AddNewFeatureImgclassification_visionmodel

Signed-off-by: swathi <konakanchi.swathi@gmail.com>

KonakanchiSwathi committed:
Update __init__.py

Deleted #imageclassificationpipeline in exclusion_pipeline_types  and also deleted the code 
    impermissible_attrs={""}
    for attr in impermissible_attrs:
        if getattr(pipeline, attr, None) is not None:
            return False

Signed-off-by: Konakanchi Swathi <98085410+KonakanchiSwathi@users.noreply.github.com>

KonakanchiSwathi committed:
Update mlflow/transformers/__init__.py

tested locally

Co-authored-by: Ben Wilson <39283302+BenWilson2@users.noreply.github.com>
Signed-off-by: Konakanchi Swathi <98085410+KonakanchiSwathi@users.noreply.github.com>

KonakanchiSwathi committed:
Update __init__.py

deleted code _validate_image_input_uri_or_file and used _validate_str_input_uri_or_file to images also

Signed-off-by: Konakanchi Swathi <98085410+KonakanchiSwathi@users.noreply.github.com>

KonakanchiSwathi committed:
Update __init__.py

intended

Signed-off-by: Konakanchi Swathi <98085410+KonakanchiSwathi@users.noreply.github.com>

KonakanchiSwathi committed:
Update tests/test_Imageclassification.py

noted will follow

Co-authored-by: Ben Wilson <39283302+BenWilson2@users.noreply.github.com>
Signed-off-by: Konakanchi Swathi <98085410+KonakanchiSwathi@users.noreply.github.com>

KonakanchiSwathi committed:
Update test_Imageclassification.py

made changes in functions also Model_name to model_name

Signed-off-by: Konakanchi Swathi <98085410+KonakanchiSwathi@users.noreply.github.com>

KonakanchiSwathi committed:
added_testcases_for_imagecalssification

BenWilson2 commented on tests/transformers/test_transformers_model_export.py:
```diff
@@ -2056,6 +2127,47 @@ def test_qa_pipeline_pyfunc_predict(small_qa_pipeline):
+        model_uri = mlflow.get_artifact_uri(artifact_path)
+
+    # Define the image file path or data
+    image_file_path = "https://raw.githubusercontent.com/mlflow/mlflow/master/tests/datasets/cat.png"  # Replace with the actual image file path
```
Could all of these in-line comments be removed? 

As a general rule, we only add Nota bene (NB) comments in test suites, since it's only developers who are reading unit tests. All of the comments in these tests can be removed.

KonakanchiSwathi replied:
done

BenWilson2 commented on tests/transformers/test_transformers_model_export.py:
```diff
@@ -3787,3 +3947,69 @@ def predict(self, context, model_input, params=None):
             )
+
+
+def test_model_distributed_across_devices():
```
instead of merging master into your branch, could you rebase? These test additions are from another PR.

KonakanchiSwathi replied:
Yes added https://github.com/mlflow/mlflow/pull/10041

KonakanchiSwathi committed:
Update test_transformers_model_export.py

Removed the comments 

Signed-off-by: Konakanchi Swathi <98085410+KonakanchiSwathi@users.noreply.github.com>

KonakanchiSwathi committed:
Update test_Imageclassification.py

changed url to https://raw.githubusercontent.com/mlflow/mlflow/master/tests/datasets/cat.png

Signed-off-by: Konakanchi Swathi <98085410+KonakanchiSwathi@users.noreply.github.com>

KonakanchiSwathi committed:
Merge branch 'master' into Added_testcases_imageclassification_model

Signed-off-by: Konakanchi Swathi <98085410+KonakanchiSwathi@users.noreply.github.com>

KonakanchiSwathi commented:
added to other branch
